lib/origen_testers/igxl_based_tester/base.rb

```diff
@@ -717,7 +717,11 @@ def pattern_footer(options = {})
         $tester.align_to_last
         # cycle(:microcode => "#{$dut.end_of_pattern_label}:") if $dut.end_of_pattern_label
         if options[:end_in_ka]
+          if $tester.j750?
```

Comment:
It does not assume that they are the same.  The base is where they are in common, i.e they happen to be the same.  If different is needed then will need to describe them generically in base (say a method or option) and differentiate how those get defined in classes that will inherit from base.  Here the microcode difference could be called @keepalive as used in base where that is defined in j750.rb and ultraflex.rb.  Compare the initialize of those files to see other diffs.